django/utils/log.py

```diff
@@ -92,6 +92,13 @@ def __init__(self, include_html=False, email_backend=None, reporter_class=None):
         )
 
     def emit(self, record):
+        # Shortcut when no email will be sent
+        if (
+            not settings.ADMINS
+            # Method not overridden
+            and self.send_mail.__func__ is AdminEmailHandler.send_mail
```

Comment:
This check is necessary in case the user has a custom `AdminEmailHandler` with an overridden `send_mail` function, [as documented](https://docs.djangoproject.com/en/5.0/ref/logging/#django.utils.log.AdminEmailHandler.send_mail), since we don’t know what sending function the custom method will call. I guess that in that case, users are responsible for performance.

Reply:
The strict reviewer in me would say this needs a test 🤷

Reply:
We have coverage with this test [`logging_tests.tests.AdminEmailHandlerTest.test_customize_send_mail_method`](https://github.com/django/django/blob/c223d14025dd9ef0d354332c537ed8622a1ec29c/tests/logging_tests/tests.py#L398) 👍